reference-implementation/lib/helpers.js

```diff
@@ -62,6 +62,10 @@ exports.IsFiniteNonNegativeNumber = v => {
 };
 
 exports.InvokeOrNoop = (O, P, args) => {
+  assert(exports.typeIsObject(O) === true);
+  assert(typeof P === 'string');
```

Comment:
This is already in the spec, sort of, as "Assert: ! IsPropertyKey(P) is true." Probably there should be a (non-exported?) helper for that in this file, which checks if typeof is string or symbol.